### ISSUE TITLE
关于测试时网络输出结果out后续的数据处理

### ISSUE THREAD

**Issue by MengFanZe1** [cfzd/Ultra-Fast-Lane-Detection#227]
您好，我主要想问一下demo.py中的这部分代码
            **with torch.no_grad():
                out = net(imgs)

            col_sample = np.linspace(0, 800 - 1, cfg.griding_num)
            col_sample_w = col_sample[1] - col_sample[0]

            out_j = out[0].data.cpu().numpy()
            out_j = out_j[:, ::-1, :]
            prob = scipy.special.softmax(out_j[:-1, :, :], axis=0)
            idx = np.arange(cfg.griding_num) + 1
            idx = idx.reshape(-1, 1, 1)
            loc = np.sum(prob * idx, axis=0)
            out_j = np.argmax(out_j, axis=0)
            loc[out_j == cfg.griding_num] = 0
            out_j = loc**
经过调试，我感觉可以直接通过out_j的输出去得到大致的最符合那一类别车道线的grid呀，为什么还要再softmax处理后再相乘求和呢？以及这一步 out_j = out_j[:, ::-1, :]，为什么要步距为-1反转呢？以及为什么prob要乘以对应的grid索引呢？
问题比较多，可否简单梳理讲解一下你这部分数据处理的思路呢？
万分期待您的解答。

**Comment by cfzd** [comment]
@MengFanZe1 
是的，argmax直接去得到也可以获得位置，不过可以看一下我们论文中的表3，argmax会比求数学期望效果差一些。这主要是因为我们这个任务的分类是一个贯序分类任务，不同类别之间有着天然的顺序关系，但是普通分类是没有这一属性的。所以普通分类用argmax，我们使用期望。至于为什么要反转，其实不反转也可以，只要相应的y坐标（也就是row anchor的位置）的顺序跟着变就行了。最后一个为什么要乘以grid索引，因为这就是计算数学期望的公式：sum(p_x * f(x))

**Comment by MengFanZe1** [comment]
理解了，万分感谢！！！
